api/search.go

```diff
@@ -132,3 +140,37 @@ func SearchHandlerFunc(queryBuilder QueryBuilder, elasticSearchClient ElasticSea
 
 	}
 }
+
+func CreateSearchIndexHandlerFunc(elasticSearchClient ElasticSearcher) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
+		indexName := createIndexName("ons")
+		fmt.Printf("Index created: %s\n", indexName)
+
+		indexCreated, err := elasticSearchClient.CreateNewEmptyIndex(ctx, indexName)
+		if !indexCreated {
+			if err != nil {
+				http.Error(w, serverErrorMessage, http.StatusInternalServerError)
+			}
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusCreated)
+		createIndexResponse := CreateIndexResponse{
+			IndexName: indexName}
+		jsonResponse, _ := json.MarshalIndent(createIndexResponse, "", " ")
+
+		_, err = w.Write(jsonResponse)
+		if err != nil {
+			log.Error(ctx, "writing response failed", err)
+			http.Error(w, serverErrorMessage, http.StatusInternalServerError)
+			return
+		}
+	}
+}
+
+func createIndexName(s string) string {
+	now := time.Now()
+	return fmt.Sprintf("%s%d", s, now.UnixMicro())
```

Comment:
This will be unique but it's not exactly human readable. Not sure how this will be used but do you think having human readable index names would be useful/valuable? If so consider applying a data/time format.

Reply:
I did check this name with Nathan (my tech lead) earlier as it is what I was told to use. The resulting index names are like this: ons1636983799453041. The timestamp in the index name is the time since epoch. The idea is that the name is not likely to conflict if there are multiple reindexes occurring at the same time.